src/components/ars.js

```diff
@@ -1,105 +1,132 @@
 /**
  * Ars
  * The main element for Ars Arsenal
+ * @flow
  */
 
-import Picker from './picker'
 import React from 'react'
+import cx from 'classnames'
+import Picker from './picker'
 import Selection from './selection'
 import MultiSelection from './multiselection'
-import Sync from '../mixins/sync'
-import createClass from 'create-react-class'
-import cx from 'classnames'
-import { func } from 'prop-types'
-
-let Ars = createClass({
-  mixins: [Sync],
-
-  propTypes: {
-    onChange: func
-  },
-
-  getDefaultProps() {
-    return {
-      onChange: () => {},
-      rootAttributes: {},
-      multiselect: false,
-      resource: 'Photo',
-      mode: 'gallery'
-    }
-  },
+import { type Record } from '../record'
+
+type Props = {
+  makeQuery: string => string,
+  makeURL: (string, any) => string,
+  rootAttributes: { [string]: * },
+  multiselect: boolean,
+  resource: string,
+  mode: 'gallery' | 'table',
+  picked: string | number | Array<string | number>,
+  columns?: string[],
+  url: string,
+  onFetch: (*) => Record,
+  onError: (*) => *,
+  onChange: (*) => any
+}
+
+type State = {
+  dialogOpen: boolean,
+  picked: Array<string | number>
+}
+
+export default class Ars extends React.Component<Props, State> {
+  static defaultProps = {
+    makeQuery: (query: string) => `q=${query}`,
+    makeURL: (url: string, id: ?mixed) => url + (id ? '/' + String(id) : ''),
+    onError: (response: *) => response,
+    onFetch: (data: *) => data,
+    onChange: () => {},
+    rootAttributes: {},
+    multiselect: false,
+    resource: 'Photo',
+    mode: 'gallery',
+    url: ''
+  }
 
-  getInitialState() {
-    let { picked } = this.props
-    if (picked && !Array.isArray(picked)) {
-      picked = [picked]
-    }
+  constructor(props: Props, context: *) {
+    super(props, context)
 
-    return {
+    this.state = {
       dialogOpen: false,
-      picked: picked
+      picked: [].concat(props.picked || [])
     }
-  },
+  }
+
+  syncProps() {
+    let { makeURL, makeQuery, onError, onFetch, url } = this.props
+    return { makeURL, makeQuery, onError, onFetch, url }
+  }
 
   getPicker() {
     let { picked } = this.state
     let { columns, multiselect, mode } = this.props
 
     return (
       <Picker
-        key="dialog"
-        ref="picker"
         {...this.syncProps()}
```

Comment:
out of curiosity, where was `syncProps()` defined before this PR? looks like it was added here but this call existed before

Reply:
it came from `mixins: [Sync]`

Reply:
It was defined in the Sync mixin:
https://github.com/vigetlabs/ars-arsenal/blob/master/src/mixins/sync.js

Long term, I think the best plan for this sort of thing is to expose a configuration object using context. That'll avoid needing to pass around a bunch of props everywhere that aren't being used. 

That's coming up pretty soon!